packages/format-library/src/link/inline.js

```diff
@@ -48,7 +49,18 @@ function InlineLinkUI( {
 		...nextLinkValue,
 	};
 
+	function removeLink() {
+		const newValue = removeFormat( value, 'core/link' );
+		onChange( newValue );
+		stopAddingLink();
+		speak( __( 'Link removed.' ), 'assertive' );
+	}
+
 	function onChangeLink( nextValue ) {
+		if ( null === nextValue ) {
+			return removeLink();
+		}
```

Comment:
Why would we need this?

Reply:
I mean...I have no use case. I guess we can remove to keep this PR focused.

Reply:
I removed to keep PR focused.